pkg/util/export/batch_processor.go

```diff
@@ -91,8 +91,8 @@ func (b *bufferHolder) Add(item batchpipe.HasName) {
 		time.Sleep(time.Millisecond)
 		b.mux.RLock()
 	}
-	defer b.mux.RUnlock()
 	b.buffer.Add(item)
+	b.mux.RUnlock()
```

Comment:
Since you are modifying the buffer, why not simply Lock() instead of RLock()?

Reply:
use RLock() should have better proformance when buffer support lock-free insertion.  although `buffer` doesn't support it yet.

Reply:
i will simply use Lock().
obviously, `append(slice, elem)` has a good enough performance in this situation, which is  6,920,719 QPS in one goroutinue with M1 cpu.